lib/LaTeXML/Util/Pathname.pm

```diff
@@ -46,6 +46,20 @@ our @EXPORT = qw( &pathname_find &pathname_findall &pathname_kpsewhich
   &pathname_cwd &pathname_chdir &pathname_mkdir &pathname_copy
   &pathname_installation);
 
+my $ISWINDOWS;
+
+BEGIN {
+  $ISWINDOWS = $^O =~ /^(MSWin|NetWare|cygwin)/i;
+  require Win32::ShellQuote if $ISWINDOWS;
+}
+
+# on Windows, command arguments are joined in a space-separated list to be parsed by CommandLineToArgvW
+# thus calling open FILEHANDLE,MODE,EXPR,LIST is not safe in the presence of special characters
+# Win32::ShellQuote::quote_system_list escapes the arguments according to CommandLineToArgvW
+# WARNING: per documentation, 'you must ensure you have more than one item being quoted for the list to be usable'
```

Comment:
Have we ensured that? I'm unclear what this warning means. Should we only call `quote_system_list` when `scalar(@_) > 1` or ...?

Reply:
> Should we only call `quote_system_list` when `scalar(@_) > 1` or ...?

Yes. This is a maddening Perl quirk. To trigger the safe call you must use `open FILEHANDLE,MODE,EXPR,LIST`, in which case `EXPR` is treated as the binary file name, no parsing.

However, if `LIST` is empty, the call becomes `open FILEHANDLE,MODE,EXPR` and Perl uses heuristics to decide if `EXPR` is a command line (e.g. contains `>/dev/null`) and possibly runs a shell on it.

So safe calls are guaranteed only when `(EXPR,LIST)` has length at least 2. At least that's what the Win32::ShellQuote author says: https://metacpan.org/release/HAARG/Win32-ShellQuote-0.003001/view/lib/Win32/ShellQuote.pm#quote_system_list

Reply:
OK, I think I understand; the comments are somewhat confusing out of context.   So, basically, we must have both `$kpsewhich` and `@candidates` to get the proper call.  And in fact we check that in the 2nd line of `pathname_kpsewhich`;  Although arguably that line might be safer in hindsight as
```
return unless $kpsewhich && grep { $_; } @candidates;
```
And moreover this may be a case where it's clearer *not* to pull `pathname_quote_system_list` out as a separate sub, but embed it within `pathname_kpsewhich`.  (so nobody's tempted to export it and call with less tested args).

Reply:
Right, I moved the code around and updated the comments, hopefully for the better. The 2nd line of `pathname_kpsewhich` is ok: for the purposes of calling `open` safely, `return unless @candidates` would already be enough (that is to say, empty or even undefined arguments are fine, it's all about the length of `@candidates`).